### ISSUE TITLE
[LowerToHW] LowerToHW Can't Handle 0-width Constants

### ISSUE THREAD

**Issue by seldridge** [llvm/circt#2269]
The following is crashing on an old assertion inside LowerToHW:

```mlir
firrtl.circuit "top_mod"   {
  firrtl.module @top_mod(out %a: !firrtl.uint<0>) {
    %c0_ui0 = firrtl.constant 0 : !firrtl.uint<0>
    firrtl.connect %a, %c0_ui0 : !firrtl.uint<0>, !firrtl.uint<0>
  }
}
```

With the change in https://github.com/llvm/circt/pull/2268, this type of code can now be generated and we need to lower it.  Specifically, the above is reduced from the nightly performance regression `test1.fir` with #2268 applied.  The above code will then fail on `main` without #2268.

**Comment by lattner** [comment]
I'll take this now.

**Comment by lattner** [comment]
fixed in 97da1889
